src/include/flockmtl/core/functions/aggregate/llm_rerank.hpp

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include "flockmtl/common.hpp"
+#include <nlohmann/json.hpp>
+
+namespace flockmtl {
+namespace core {
+
+class LlmReranker {
+public:
+    string model;
+    int model_context_size;
+    string user_prompt;
+    string llm_reranking_template;
+    int batch_size;
+
+    LlmReranker(vector<nlohmann::json>& tuples, string& model, int model_context_size, string& user_prompt, string& llm_reranking_template);
+
+    vector<int> LlmRerank(vector<pair<int, nlohmann::json>> tuples);
+
+    vector<nlohmann::json> SlidingWindowRerank(vector<nlohmann::json>& tuples);
```

Comment:
this is a private method used in LlmRerank, it should be `LlmRerankWithSlidingWindow`

Reply:
done